v2/controllers/marketplace/certissuer_controller.go

```diff
@@ -0,0 +1,179 @@
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+	for _, cm := range configMapList.Items {
+		if _, ok := cm.Annotations["service.beta.openshift.io/inject-cabundle"]; ok {
+			if _, ok := cm.Data["service-ca.crt"]; ok {
+				err := r.InjectCACertIntoConfigMap(&cm)
+				if err != nil {
+					log.Error(err, "failed to inject CA certificate")
+				}
+			} else {
+				klog.V(1).Info("service-ca.crt field was not found in a configmap labeled with inject-cabundle")
+			}
+		}
+	}
+
+	// Create missing TLS secrets
+	tlsSecrets := []string{
+		"prometheus-operator",
+		"rhm-metric-state",
+		"rhm-prometheus-meterbase",
+	}
+
+	err = r.CreateTLSSecrets(req.NamespacedName.Namespace, tlsSecrets...)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *CertIssuerReconciler) Inject(injector *inject.Injector) inject.SetupWithManager {
+	injector.SetCustomFields(r)
+	return r
+}
+
+func (r *CertIssuerReconciler) InjectCertIssuer(ci *utils.CertIssuer) error {
+	r.certIssuer = ci
+	return nil
+}
+
+func (r *CertIssuerReconciler) InjectOperatorConfig(oc *config.OperatorConfig) error {
+	r.cfg = oc
+	return nil
+}
+
+// InjectCACertIntoConfigMap injects certificate data into
+func (r *CertIssuerReconciler) InjectCACertIntoConfigMap(configmap *corev1.ConfigMap) error {
+	cm := configmap
+	patch := client.MergeFrom(cm.DeepCopy())
+	cm.Data["service-ca.crt"] = string(r.certIssuer.CAPublicKey())
+	return r.Client.Patch(context.Background(), cm, patch)
+}
+
+// CreateTLSSecrets creates tls secrets signed by recociler's CA
+func (r *CertIssuerReconciler) CreateTLSSecrets(namespace string, secretNames ...string) error {
+	cert, key, err := r.certIssuer.CreateCertFromCA(types.NamespacedName{
+		Name:      secretNames[0],
+		Namespace: namespace,
+	})
+	if err != nil {
+		return err
+	}
+	secretName := fmt.Sprintf("%s-tls", secretNames[0])
+	secretMeta := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      secretName,
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			"tls.crt": cert,
+			"tls.key": key,
+		},
+	}
+
+	err = r.Client.Create(context.Background(), secretMeta)
+	if err != nil {
+		r.Log.Error(err, "cannot create tls secret")
+	}
+
+	r.Log.Info("created tls secret", "secretName", secretName)
+
+	if len(secretNames) == 1 {
+		return nil
+	}
+
+	return r.CreateTLSSecrets(namespace, secretNames[1:]...)
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *CertIssuerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if r.cfg.Infrastructure.HasOpenshift() {
+		return nil
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&marketplaceredhatcomv1beta1.CertIssuer{}).
```

Comment:
Not using CertIssuer here. You're really watching for the configmaps. We can remove the CRD and just watch for those configmaps/secrets.